### DISCUSSION TITLE
Why not call `LetsPlot.setup_html()` automatically upon importing lets_plot

### DISCUSSION THREAD

**Discussion by Zaf4** [JetBrains/lets-plot#928] [General]
First of all, I'd like to thank you for your great effort and determination on this project. It helps fill a major void in python ecosystem that is data visualization. Of course, there are some libraries such as seaborn but they don't follow the layered principle of ggplot. Having used seaborn in my last publication, lack of modularity and bad aesthetics had me overworked for making even simple graphs to look good. Lets_plot graphs, like ggplot2, look great out-of-the-box and highly modular and customizable as its R counterpart. So, thank you again. 

With that, there are simple things that could be done to make lets_plot much more attractive.

1. As in the title, it is a single line but having to write `LetsPlot.setup_html()` everytime is an inconvenience. So, why importing the lets_plot not call the function directly?
2. There is also lack of scientific formating (10<sup>n</sup> and 2<sup>n</sup>) of axis for logaritmic scales.

These changes may boost the use of lets_plot in science and in general.

**Comment by alshan** [comment]
Hi!
"scientific formatting" is currently among high-priority tasks and you can expect it by the end of the year. 

---
Why not call LetsPlot.setup_html() automatically upon importing lets_plot.

Maybe it would be ok but maybe not.
I see explicit invocation of `LetsPlot.setup_html()` as a safer way to inject HTML/JS into a notebook. 

Concerns about  calling `LetsPlot.setup_html()` on module initialization: 

Lifecycle of a python module is under control of the python interpreter whereas lifecycle of a cell (namely, an output cell with the injected HTML/JS inside) is under control of the "notebook".
For example, if you delete a cell containing `from lets_plot import *` , then add `from lets_plot import *` to some other cell and re-evaluate , than `setup_html()` wont be invoked (because module is already loaded) and I can't say for sure what will happen to charts.

On the other hand. there are might be use-cases where invoking `setup_html()` would be rather an undesirable side-effect, for example: https://github.com/holoviz/panel/issues/5475#issuecomment-1794208653
